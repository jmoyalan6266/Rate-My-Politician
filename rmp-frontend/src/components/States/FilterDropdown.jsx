import { DropdownButton, Dropdown, FormControl } from "react-bootstrap";

const FilterDropdown = ({ filterBy, setFilterBy, filter, setFilter }) => {
    return (
        <>
            <DropdownButton
                id="dropdown-basic-button"
                title={filterBy}
                className="m-3"
            >
                <Dropdown.Item onClick={() => setFilterBy("Name")}>
                    Name
                </Dropdown.Item>
                <Dropdown.Item onClick={() => setFilterBy("Population")}>
                    Population
                </Dropdown.Item>
                <Dropdown.Item onClick={() => setFilterBy("Median Income")}>
                    Median Income
                </Dropdown.Item>
                <Dropdown.Item onClick={() => setFilterBy("Poverty Rate")}>
                    Poverty Rate
                </Dropdown.Item>
                <Dropdown.Item
                    onClick={() => setFilterBy("2016 Election Margin")}
                >
                    2016 Election Margin
                </Dropdown.Item>
            </DropdownButton>
            <FormControl
                type="text"
                placeholder="Filter"
                value={filter}
                onChange={(event) => setFilter(event.target.value)}
                className="my-3"
                style={{ maxWidth: "10rem" }}
            />
        </>
    );
};

export default FilterDropdown;
