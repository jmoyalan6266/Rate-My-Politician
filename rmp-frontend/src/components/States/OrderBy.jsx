import { MdArrowUpward, MdArrowDownward } from "react-icons/md";

const OrderBy = ({ ascending, setAscending, allStates, setAllStates }) => {
    if (ascending) {
        return (
            <MdArrowUpward
                className="my-3"
                size="2rem"
                onClick={() => {
                    setAscending(false);
                    setAllStates(allStates.reverse());
                }}
            />
        );
    } else {
        return (
            <MdArrowDownward
                className="my-3"
                size="2rem"
                onClick={() => {
                    setAscending(true);
                    setAllStates(allStates.reverse());
                }}
            />
        );
    }
};

export default OrderBy;
