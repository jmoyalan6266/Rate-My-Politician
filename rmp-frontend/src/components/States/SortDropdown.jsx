import { DropdownButton, Dropdown } from "react-bootstrap";

const SortDropdown = ({ sortBy, setSortBy, allStates, setAllStates, setAscending }) => {
    const handleNameSort = () => {
        setSortBy("Name");
        setAllStates(allStates.sort((a, b) => (a.name > b.name ? 1 : -1)));
        setAscending(true);
    };

    const handlePopSort = () => {
        setSortBy("Population");
        setAllStates(allStates.sort((a, b) => a.population - b.population));
        setAscending(true);
    };

    const handleIncSort = () => {
        setSortBy("Median Income");
        setAllStates(
            allStates.sort((a, b) => a.median_income - b.median_income)
        );
        setAscending(true);
    };

    const handlePovSort = () => {
        setSortBy("Poverty Rate");
        setAllStates(allStates.sort((a, b) => a.poverty_rate - b.poverty_rate));
        setAscending(true);
    };

    const handleElecSort = () => {
        setSortBy("2016 Election Margin");
        setAllStates(allStates.sort((a, b) => a.dem_margin - b.dem_margin));
        setAscending(true);
    };

    return (
        <DropdownButton
            id="dropdown-basic-button"
            title={sortBy}
            className="m-3"
        >
            <Dropdown.Item onClick={handleNameSort}>Name</Dropdown.Item>
            <Dropdown.Item onClick={handlePopSort}>Population</Dropdown.Item>
            <Dropdown.Item onClick={handleIncSort}>Median Income</Dropdown.Item>
            <Dropdown.Item onClick={handlePovSort}>Poverty Rate</Dropdown.Item>
            <Dropdown.Item onClick={handleElecSort}>
                2016 Election Margin
            </Dropdown.Item>
        </DropdownButton>
    );
};

export default SortDropdown;
