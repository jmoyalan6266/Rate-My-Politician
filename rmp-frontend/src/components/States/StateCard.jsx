import { Card, Badge, Button } from 'react-bootstrap';
import { Link } from "react-router-dom";
import { numToString, parseElectionMargin, getHighlightedText } from "../../utils";

const StateCard = ({ state, query }) => {
    return (
        <Card style={{ width: "20rem" }} className="h-100">
            <Card.Img variant="top" src={state.map_image_url} />
            <Card.Body>
                <center>
                    <Card.Title>{getHighlightedText(state.name, query)}</Card.Title>
                    <Card.Text>
                        <strong>Population:</strong> {getHighlightedText(numToString(state.population), query)}
                    </Card.Text>
                    <Card.Text>
                        <strong>Median Income:</strong> $
                        {getHighlightedText(numToString(state.median_income), query)}
                    </Card.Text>
                    <Card.Text>
                        <strong>Poverty Rate:</strong>{" "}
                        {getHighlightedText(Number((state.poverty_rate * 100).toFixed(1)), query)}%
                    </Card.Text>
                    <Card.Text>
                        <strong>2016 Election Margin:</strong>{" "}
                        {parseElectionMargin(state.dem_margin).includes("D") ? (
                            <Badge pill variant="primary">
                                {parseElectionMargin(state.dem_margin)}
                            </Badge>
                        ) : (
                            <Badge pill variant="danger">
                                {parseElectionMargin(state.dem_margin)}
                            </Badge>
                        )}
                    </Card.Text>
                    <Button as={Link} to={`/states/${state.index}`}>
                        More Info
                    </Button>
                </center>
            </Card.Body>
        </Card>
    );
};

export default StateCard;
