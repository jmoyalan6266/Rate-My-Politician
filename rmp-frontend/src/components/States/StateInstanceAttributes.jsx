import { Row, Image, Container, Card, Badge } from "react-bootstrap";
import twitter from "./twitter_logo.png";
import { numToString, parseElectionMargin } from "../../utils";

import { Link } from "react-router-dom";

const StateInstanceAttributes = ({ state }) => {
    return (
        <Container className="mt-5">
            <Row>
                <Card className="mx-auto m-3">
                    <Card.Header>Median Property Value</Card.Header>
                    <Card.Body className="mx-auto">
                        <Card.Text>
                            ${numToString(state.property_value)}
                        </Card.Text>
                    </Card.Body>
                </Card>
                <Card className="mx-auto m-3">
                    <Card.Header>Median Age</Card.Header>
                    <Card.Body className="mx-auto">
                        <Card.Text>{state.median_age}</Card.Text>
                    </Card.Body>
                </Card>
                <Card className="mx-auto m-3">
                    <Card.Header>Number of US Representatives</Card.Header>
                    <Card.Body className="mx-auto">
                        {state.num_representatives}
                    </Card.Body>
                </Card>
                <Card className="mx-auto m-3">
                    <Card.Header>2016 Election Margin</Card.Header>
                    <Card.Body className="mx-auto">
                        {parseElectionMargin(state.dem_margin).includes("D") ? (
                            <Badge pill variant="primary">
                                {parseElectionMargin(state.dem_margin)}
                            </Badge>
                        ) : (
                            <Badge pill variant="danger">
                                {parseElectionMargin(state.dem_margin)}
                            </Badge>
                        )}
                    </Card.Body>
                </Card>
                <Card className="mx-auto m-3">
                    <Card.Header>Chronically Homeless Individuals</Card.Header>
                    <Card.Body className="mx-auto">
                        {numToString(state.chronically_homeless)}
                    </Card.Body>
                </Card>
                <Card className="mx-auto m-3">
                    <Card.Header>Diabetes Prevalence</Card.Header>
                    <Card.Body className="mx-auto">
                        {Number((state.diabetes_prevalence * 100).toFixed(1))}%
                    </Card.Body>
                </Card>
                <Card className="mx-auto m-3">
                    <Card.Header>Twitter</Card.Header>
                    <Card.Body className="mx-auto">
                        <a href={state.twitter}>
                            <Image
                                src={twitter}
                                style={{ maxHeight: "1.5rem" }}
                            />
                        </a>
                    </Card.Body>
                </Card>
            </Row>
        </Container>
    );
};

export default StateInstanceAttributes;
