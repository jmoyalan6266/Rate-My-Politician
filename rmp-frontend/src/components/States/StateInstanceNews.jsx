import { Row, Col, Card, Button, Container} from "react-bootstrap";
import { Link } from "react-router-dom";
import useAxios from "axios-hooks";
import { IoIosArrowBack, IoIosArrowForward } from "react-icons/io";
import { useState } from "react";

const StateInstanceNews = ({ state }) => {
    const [{ data, loading, error }, refetch] = useAxios(
        `${process.env.REACT_APP_API_URL}/states/${state.name}/news`
    );
    const [index, setIndex] = useState(0);

    return (
        <Container className="m-5">
        <Col>
            <center>
                <h1>News Articles</h1>
            </center>
            <Row>
                <IoIosArrowBack
                    style={{ marginTop: "15rem" }}
                    onClick={() =>
                        index > 0 ? setIndex(index - 1) : null
                    }
                    size="2.5em"
                />
                {data?.articles.slice(index, index + 4).map((article, i) => (
                    <Col style={{width: "10%"}} key={i}>
                        <Card
                            key={i}
                            style={{ width: "100%" }}
                            className="mx-auto m-4"
                        >
                            <center>
                                <Card.Header>
                                    <Link to={`/news/${article.source.name}`}>
                                        <strong>{article.source.name}</strong>
                                    </Link>
                                </Card.Header>

                                <Card.Img src={article.urlToImage} variant="top" />
                                <Card.Body>
                                    <Card.Text className="p-3">
                                        {article.title}
                                    </Card.Text>
                                    <Button as="a" href={article.url} size="sm">
                                        Article Link
                                    </Button>
                                </Card.Body>
                            </center>
                        </Card>
                    </Col>
                ))}
                <IoIosArrowForward
                    style={{ marginTop: "15rem" }}
                    onClick={() =>
                        index < 19 ? setIndex(index + 1) : null
                    }
                    size="2.5em"
                />
            </Row>
        </Col>
        </Container>
    );
};

export default StateInstanceNews;
