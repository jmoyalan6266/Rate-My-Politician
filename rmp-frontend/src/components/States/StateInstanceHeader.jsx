import { Row, Image, Jumbotron } from "react-bootstrap";
import StateInstanceAttributes from "./StateInstanceAttributes";

const StateInstanceHeader = ({ state }) => {
    const imageType = Math.random() > 0.5 ? "landscape" : "skyline";

    return (
        <div>
            <h1 className="my-4">{state.name}</h1>
            <Jumbotron
                style={{
                    backgroundImage: `url(${
                        imageType === "landscape"
                            ? state.landscape_background_url
                            : state.skyline_background_url
                    })`,
                    height: "100%",
                }}
            >
                <Row>
                    <Image
                        src={state.state_flag_url}
                        style={{ maxHeight: "12rem" }}
                        className="mx-auto"
                    />
                    <Image
                        src={state.state_seal_url}
                        style={{ maxHeight: "12rem" }}
                        className="mx-auto"
                    />
                </Row>
                <StateInstanceAttributes state={state} />
            </Jumbotron>
        </div>
    );
};

export default StateInstanceHeader;
