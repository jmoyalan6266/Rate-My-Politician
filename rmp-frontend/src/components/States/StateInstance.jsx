import {
    Container,
    Row,
} from "react-bootstrap";
import { useParams } from "react-router";
import useSingleState from "../../hooks/useSingleState";
import Loading from "../Loading";
import StateInstanceHeader from "./StateInstanceHeader";
import StateInstancePoliticians from "./StateInstancePoliticians";
import StateInstanceNews from "./StateInstanceNews";

const StateInstance = () => {
    const { id } = useParams();
    const { state, isLoading } = useSingleState(id);

    return (
        <Container className="my-4">
            {isLoading ? (
                <Loading />
            ) : (
                <div>
                    <center>
                        <StateInstanceHeader state={state} />
                    </center>
                    <Row>
                        <StateInstancePoliticians state={state} />
                    </Row>
                    <Row>
                        <StateInstanceNews state={state} />
                    </Row>
                </div>
            )}
        </Container>
    );
};

export default StateInstance;
