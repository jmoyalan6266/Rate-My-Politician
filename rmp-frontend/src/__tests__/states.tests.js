import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from "enzyme-adapter-react-16"
import States from "../components/States/StateModel"

configure({ adapter: new Adapter() })

describe("Test State Model page views", () => {
	test("States", () => {
		const statesTest = shallow(<States />)
		expect(statesTest).toMatchSnapshot()
	})
})