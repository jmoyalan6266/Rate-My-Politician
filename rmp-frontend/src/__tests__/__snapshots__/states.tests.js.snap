// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test State Model page views States 1`] = `
<Container
  className="my-4"
  fluid={false}
>
  <StateModelHeader
    allStates={Array []}
    ascending={true}
    filter=""
    filterBy="Name"
    setAllStates={[Function]}
    setAscending={[Function]}
    setFilter={[Function]}
    setFilterBy={[Function]}
    setSortBy={[Function]}
    sortBy="Name"
  />
  <Row
    noGutters={false}
  />
  <StatePagination
    allStates={Array []}
    page={1}
    setPage={[Function]}
  />
</Container>
`;
